docs/beman-standard.md

```diff
@@ -0,0 +1,170 @@
+recommendations.
+
+* **Rules** must be followed in order to conform to this standard. These entries
+  are prefixed by "RULE:".
+* **Recommendations** should be followed in general, but specific circumstances
+  may make this a less-than-ideal choice. Libraries not following a specific
+  recommendation can still conform to this standard. These entries are prefixed
+  by "RECOMMENDATION:".
+
+## License
+
+**[LICENSE.APPROVED]** REQUIREMENT: All Beman libraries must be licensed
+under an approved license. These are:
+
+1. [Apache License v2.0 with LLVM Exceptions](https://llvm.org/LICENSE.txt)
+2. [Boost Software License 1.0](https://www.boost.org/LICENSE_1_0.txt)
+3. [The MIT License](https://opensource.org/license/mit)
+
+**[LICENSE.BOOST]** RECOMMENDATION: All Beman libraries should be licensed under
+the [Apache License v2.0 with LLVM Exceptions](https://llvm.org/LICENSE.txt).
+
+**[LICENSE.CRITERIA]** REQUIREMENT: All approved licenses must meet the
+following requirements:
+
+1. Simple to read and understand.
+2. Permission without fee to copy, use and modify the software for any
+   use (commercial and non-commercial).
+3. Requires that the license appears on all copies of the software source code.
+4. Must not require that the license appears with executables or other binary
+   uses of the library.
+5. Must not require that the source code be available for execution or other
+   binary uses of the library.
+
+## General
+
+**[LIBRARY_NAMES]** REQUIREMENT: Beman libraries names begin with `Beman.`
+followed by an UpperCamelCase short name.
+
+Examples are Beman.SmartPointer and Beman.SenderReceiver.
+
+**[REPOSITORY_NAME]** RECOMMENDATION: The repository should be named after the
+library name excluding the `Beman.` prefix.
+
+For example, a Beman.SmartPointer library's repository should be named
+`SmartPointer`.
+
+## Top-level
+
+The top-level of a Beman library repository must consist of `CMakeLists.txt`,
+`LICENSE`, and `README.md` files.
+
+**[TOPLEVEL.CMAKE]** REQUIREMENT: There must be a `CMakeLists.txt` at the repository's root
+that builds and tests (via. CTest) the library.
+
+**[TOPLEVEL.LICENSE]** REQUIREMENT: There must be a `LICENSE` at the
+repository's root with the contents of an approved license that covers the
+contents of the repository.
+
+**[TOPLEVEL.README]** REQUIREMENT: There must be a markdown-formatted
+`README.md` at the repository's root that describes the library, explains how
+to build it, and links to further documentation.
+
+## `README.md`
+
+**[README.TITLE]** RECOMMENDATION: The `README.md` should begin with a level 1
+header with the name of the library optionally followed with a ":" and short
+description.
+
+For example:
+
+```markdown
+# Beman.SenderReceiver: Scalable Asychronous Program Building Blocks
+```
+
+**[README.PURPOSE]** RECOMMENDATION: Following the title, the `README.md` should
+contain a one- or two-paragraph summary describing the library's purpose.
```

Comment:
Maybe we could add something about Paper number?

```
**[README.PAPER]** RECOMMENDATION: The `README.md` should contain the reference to
the paper number the library is covering.

# Paper: p2300r4
```

Reply:
Added [here](https://github.com/beman-project/beman/pull/18/commits/0f86bf9ca3bfbfd3acb4799e355bc6cb728dcc9f), but with a bit different wording:

- Allow for multiple papers to be implemented by a single repo
- Link to the official version of the paper
- Use the paper's title and put the number in parenthesis since the latter is committee speak.